projects/observability/src/shared/components/cartesian/chart-sync-service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Band, CartesianSelectedData, Series } from '../../../public-api';
+import { MouseLocationData } from '../utils/mouse-tracking/mouse-tracking';
+
+@Injectable({ providedIn: 'root' })
+export class ChartSyncService<TData> {
+  private readonly locationChangeSubject: Subject<{
+    groupId?: string;
+    locationData: MouseLocationData<TData, Series<TData> | Band<TData>>[];
+  }> = new Subject();
+
+  public mouseLocationChange(
+    data: MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianSelectedData<TData>,
+    groupId?: string
+  ): void {
+    this.locationChangeSubject.next({
+      groupId: groupId,
+      locationData: data as MouseLocationData<TData, Series<TData> | Band<TData>>[]
+    });
+  }
+
+  public getLocationChangesForGroup(
+    _groupId: string
```

Comment:
do we need this `_groupId` ? I think groupId is still present in the `MouseLocationData`, right?